cosmos/cosmos-sdk
# remove dependencies on staking module from genutil module

## Description

Removed all references to the `staking` module from the `genutil` module

This was achieved by passing an interface to the `gentx` cobra command

and passing a function into the `collect-gentx` cobra command

~~## Updates to downstream consumers 🚨~~
~~Now when using the `CollectGenTxsCmd` and `GenTxCmd` cobra commands we need to pass in functions defined in the staking module~~

~~`gentx` => https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/simapp/simd/cmd/root.go#L93-L94 => staking.AppModuleBasic{}~~
~~`collect-gentx` => https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/simapp/simd/cmd/root.go#L91 => staking.ValidateMsgInGenesis~~

~~This means if this gets merged, downstream consumers of the sdk will have to update two lines of code outlined above in their `cmd/*d.go`~~

## How to test
Run
1. `cd /simapp/simd`
1. `go run main.go init testing --chain-id=testing`
1. `go run main.go keys add validator`
1. `go run main.go add-genesis-account validator 1000000000stake`
1. `go run main.go gentx validator --chain-id testing` => this was updated
1. `go run main.go collect-gentx` => this was updated

||

1. `make localnet-start`

I'm 1000% sure everyone knows how to test the just being through adding the `How to test` section :dancers: 

DONE: ~~Just writing tests atm, should have some soon :tennis:~~

closes: #7253 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.
